src/psyclone/domain/common/transformations/kernel_module_inline_trans.py

```diff
@@ -575,6 +575,10 @@ def apply(self, node, options=None, **kwargs):
                     newsym = container.symbol_table.lookup(code_to_inline.name)
                     newsym.visibility = Symbol.Visibility.PRIVATE
 
+            # Update any unresolved Symbols in the routine in case they are
+            # resolved in the Container scope.
+            code_to_inline.symbol_table.resolve_imports()
```

Comment:
I still don't understand why we resolve symbols that are already resolved.

If they are unresolved they shouldn't pass the validate (as we cannot prove that they match the one in the calleer scope). Why is some form of code_to_inline.replace_symbol_with not enough?